### PR TITLE
Landing page GA 

### PR DESCRIPTION
**Pull request by qcl**
Ref: #26 

Use `data-logAction` attribute in `<a>` and use `$("a").click` to bind click event. When link did clicked, it will check if there is `data-logAction`, if there is, it will log this event with `data-logAction` as action in `ga('send', 'event', Category, Action, Label, Value);`
- [x] load GA script into landing page
- [x] Tracking fb sign (in about section) in button click event &  modal-fb-sign-in button click event
- [x] Tracking submit button click event
- [x] Tracking uploading successful/failed event
- [x] Tracking checking form failed event
- [x] Tracking share button event (in share & result section)
- [x] Tracking navigation bar click event
- [x] autocomplete term 
- [x] Tracking the time of writing a form

### CODE REVIEW

**Comment by mark86092** [comment]
按下 show.html 的話，瀏覽器會等待ga送完ajax才跳轉頁面嗎？

我在想如果瀏覽器中斷請求，效果是不是追蹤不到了

**Comment by barry800414** [comment]
我剛在看official document，看起來是不是原本就有追蹤切換頁面的功能?
image

image

**Comment by barry800414** [comment]
https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits#knowing_when_the_hit_has_been_sent

彥齊你擔心的事情確實會有，這邊也有講解決方案

**Comment by barry800414** [comment]
Sorry, 我遇到幾個困難
1. 我的terminal有點問題，看到的commit message跟上傳的不一樣，我想修改，於是用了`git rebase -i HEAD~8`, 編輯器改完message後，`git commit --amend` , `git push --force-with-lease`，但看起來這邊沒有更新..
2. 目前的code還沒處理兩個問題:
   
   a. `<a>` 的預設行為如果是跳頁的話，有可能無法執行，雖然我覺得這個問題是還好。
   b. index-i13n.js如果沒有載入成功，會導致index.js無法成功執行。 我寫在index-i13n.js裡面的function可能要有更好的寫法。

**Comment by mark86092** [comment]
Actually, you don't have to wrap `sendGAEvent`, since our form is sent ajax.

**Comment by mark86092** [comment]
2.b 是什麼情況會沒辦法成功載入？

**Comment by barry800414** [comment]
> Actually, you don't have to wrap sendGAEvent, since our form is sent ajax.

這個看不太懂

**Comment by barry800414** [comment]
> 2.b 是什麼情況會沒辦法成功載入？

應該是說萬一沒有載入成功的話，原本index.js的功能也會有問題，我想說應該要先檢查一下

**Comment by mark86092** [comment]
> Actually, you don't have to wrap sendGAEvent, since our form is sent ajax.

我們的表單是透過 ajax 送出，應該不用特別 timeout 或接收 ga 的 callback，單純 `ga(xxx)` 就ok了

**Comment by barry800414** [comment]
有道理.. 我剛發現我從頭到尾沒用到callback function

**Comment by barry800414** [comment]
更新了!

**Comment by mark86092** [comment]
Look good to me

**Comment by barry800414** [comment]
- [x] load GA script into landing page
- [x] Tracking fb sign (in about section) in button click event &  modal-fb-sign-in button click event
- [x] Tracking input box on focus click event
- [x] Tracking submit button click event
- [x] Tracking uploading successful/failed event
- [x] Tracking checking form failed event
- [x] Tracking share button event (in share & result section)
- [x] Tracking navigation bar click event

**Comment by barry800414** [comment]
看request有送出去，應該沒問題。
我想說可以把所有的focus input 刪掉了，應該是沒有用途
然後有conflicts要解
